spatialdb/cube.py

```diff
@@ -233,18 +233,23 @@ def from_blosc(self, byte_arrays, time_sample_range=None):
                 # Got a list of byte arrays, so assume they are each 4-D, corresponding to time samples
 
                 # Unpack all of the arrays into the cube
-                for idx, t in enumerate(range(time_sample_range[0], time_sample_range[1])):
-                    if idx == 0:
+                b_arr_idx = 0
+                missing_gen = self.missing_ts_gen(missing_time_steps)
+                missing_t = next(missing_gen)
+                for data_idx, t in enumerate(range(time_sample_range[0], time_sample_range[1])):
+                    if data_idx == 0:
                         # On first cube get the size and allocate properly
-                        temp_mat = self.unpack_array(byte_arrays[idx], 1)
-
-                        # allocate
                         self.data = np.zeros(shape=(time_sample_range[1] - time_sample_range[0],
                                                     self.z_dim, self.y_dim, self.x_dim), dtype=self.data.dtype)
-
-                        self.data[idx, :, :, :] = temp_mat
+                    if t == missing_t:
```

Comment:
Why not just use `if t in missing_time_steps:`?

Reply:
Good question. Looking at this now, I'm not sure that it's necessary. You should have asked me a couple of months ago. 😀